mathesar_ui/src/component-library/pagination/Pagination.svelte

```diff
@@ -45,7 +45,14 @@
 
   $: pageCount = Math.ceil(total / pageSize);
   $: pageInfo = calculatePages(currentPage, pageCount);
+  let pagebutton: HTMLElement | null = null;
 
+  afterUpdate(() => {
+    pagebutton = document.querySelector(`[data-page="${currentPage}"]`);
+    if (pagebutton) {
+      pagebutton.focus();
+    }
+  });
```

Comment:
The `afterUpdate` block gets called whenever a prop's value is updated.

If a parent component changes the value, the global focus state will be set to the pagination component, which should not be the case.

The focus should only be set upon explicit user action. It is better to place this logic within the `setPage` function.

Inorder to set the focus after the DOM update, you can use the [`tick` callback](https://svelte.dev/docs#run-time-svelte-tick), within the `setPage` method.

Reply:
ok I will look into it.